samples/matter/thermostat/CMakeLists.txt

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+get_filename_component(MATTER_MODULE_ROOT $ENV{ZEPHYR_BASE}/../modules/lib/matter REALPATH)
```

Comment:
Take a look at the newest version of Matter samples in NCS - this line has been removed to remove the dependency on ZEPHYR_BASE env variable.